src/parser/mod.rs

```diff
@@ -10000,7 +10000,7 @@ impl<'a> Parser<'a> {
                     table_with_joins: Box::new(table_and_joins),
                     alias,
                 })
-            } else if dialect_of!(self is SnowflakeDialect | GenericDialect) {
+            } else if dialect_of!(self is SnowflakeDialect | GenericDialect | MsSqlDialect) {
```

Comment:
You may want to update the comments below too.

Reply:
Yeah to @lovasoa's suggestion, one double check on my end @gaoqiangz is the parenthesis around the joined table supported syntax for the openjson syntax? Thinking if its not, then we would be able to skip this change entirely. If it is then indeed lets update the comments to account for mssql.
Additionally, can we add a test case covering the behavior (as in a roundtrip test like `SELECT ... FROM T AS A CROSS APPLY (OPENJSON(...) WITH ...) AS B` so that we catch any regression on the behavior going forward

Reply:
I have checked that MSSQL doesn't support parenthesis around the openjson syntax.
I'll revert this change.